File: src/leetcode/189. Rotate Array/RotateArray189.java
class Solution {
    public void rotate(int[] nums, int k) {
        int len = nums.length;
        int[] ans = Arrays.copyOf(nums, len);

        for(int i = 0; i < len; i++) {
            nums[(i + k) % len] = ans[i];
        }
    }
}

File: src/leetcode/solved-collection/189. Rotate Array/rotate_array189.py
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        l = len(nums)
        ans = nums.copy()
        for i in range(l):
            nums[(i + k) % l] = ans[i]
